localization/ekf_localizer/src/ekf_localizer.cpp

```diff
@@ -297,12 +298,9 @@ bool EKFLocalizer::getTransformFromTF(
   tf2::BufferCore tf_buffer;
   tf2_ros::TransformListener tf_listener(tf_buffer);
   rclcpp::sleep_for(std::chrono::milliseconds(100));
-  if (parent_frame.front() == '/') {
```

Comment:
It seems a bit out of scope of this PR, but how about using this? https://github.com/autowarefoundation/autoware.universe/blob/main/common/tier4_autoware_utils/include/tier4_autoware_utils/ros/transform_listener.hpp

This getTransformFromTF function is a legacy function according to @.YamatoAndo -san, so I think we can completely replace it with the above library class.

Reply:
If we don't repeat calling lookupTransform, or if we just show the warning message and do nothing more than this, I can use the proposed function.

Reply:
@IshitaTakeshi The `EKFLocalizer::getTransformFromTF` function has some tricky code such as `rclcpp::sleep_for(std::chrono::milliseconds(100));`, so I think using the one from tier4_autoware_utils would make much more sense than leaving this function as it is. Given that this tier4_autoware_utils::transform_listener is already used in several packages, I think it is OK to replace it. (not necessarily in this PR, though)

Reply:
OK then I'm going to do it or continue the discussion on another PR so could you approve this PR first?

Reply:
OK :+1: